### ISSUE TITLE
Feature: allow embedded entities instead of resolvers

### ISSUE THREAD

**Issue by aburmeis** [arangodb/spring-data#269]
Currently any entity references from annotations like `@Relations`, `@Ref` etc. are loaded using a `RelationResolver`. This works perfectly but when fetching a lot of data, all referenced entities will be fetch with a separate call for each of them (fetching 100 entities with 5 properties referencing other entities will result in 501 calls).

For performance reasons it would be better to allow prefetching of referenced entities on server side returning a merged document:

```java
@Query("WITH `test-customer`, owns, `test-product` " +
	"FOR c IN `test-customer` FILTER c._key == @id " +
	"LET owns = FIRST(FOR p IN OUTBOUND c owns RETURN p) " +
	"LIMIT 1 " +
	"RETURN MERGE(c, { owns: owns, owns2: owns})")
Optional<Customer> findByIdUsingEmbeddedEntities(@Param("id") String id);
```

Currently the `DefaultArangoConverter` ignores such data and uses `RelationResolver`s anyway. Depending on the `VPackSlice` source, the behaviour could be adapted and either use the resolver on missing data or read the entity if data is present.

**Comment by aburmeis** [comment]
We did a patch on our side, so I would provide a PR soon.

**Comment by rashtao** [comment]
fixed by #270
